### PR TITLE
[CPM-77] create data model for student

### PR DESCRIPTION
**Pull request by HongShi10**
Created data model for student

### CODE REVIEW

File: api/src/models/student.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ *        plans:
+ *          type: Object
+ *          description: the plans that the student has
+ */
+const studentSchema = mongoose.Schema({
+    name: {
+        type: String,
+        unique: false,
+        required: true,
+    },
```
Comment (daltonlim) [inline api/src/models/student.js:31]: Should we have a student id field?

Comment (HongShi10) [reply api/src/models/student.js:31]: thats what I thought but it isnt in the data model

Comment (HongShi10) [reply api/src/models/student.js:31]: should i add it anyway?

Comment (wryao64) [reply api/src/models/student.js:31]: Yes - I'll add it to the data model

Comment (HongShi10) [reply api/src/models/student.js:34]: ok done

File: api/src/models/student.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ *    Student:
+ *      type: object
+ *      required:
+ *        - name
+ *        - upi
+ *      properties:
+ *        name:
+ *          type: string
+ *          description: name of the student
+ *        studentID:
```
Comment (wryao64) [inline api/src/models/student.js:16]: Should this be id instead? That's what it says in the class diagram

File: api/src/models/student.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+const studentSchema = mongoose.Schema({
+    name: {
+        type: String,
+        unique: false,
+        required: true,
+    },
+    studentID: {
+        type: String,
+        unique: true,
+        required: true,
```
Comment (wryao64) [inline api/src/models/student.js:38]: This isn't written as required in the swagger comment

**Review by wryao64 — CHANGES_REQUESTED** [review]

File: api/src/models/student.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+const mongoose = require("mongoose");
+const uniqueValidator = require("mongoose-unique-validator");
+/**
+ * @swagger
+ *  components:
+ *   schemas:
+ *    Student:
+ *      type: object
+ *      required:
+ *        - name
```
Comment (daltonlim) [inline api/src/models/student.js:10]: add id

File: api/src/models/student.js (hunk 2 above)
Comment (HongShi10) [reply api/src/models/student.js:16]: what do you mean?

Comment (HongShi10) [reply api/src/models/student.js:16]: oh nvm

File: api/src/models/student.js (hunk 3 above)
Comment (HongShi10) [reply api/src/models/student.js:39]: fixed

**Review by daltonlim — APPROVED** [review]
LGTM 👍

**Review by wryao64 — APPROVED** [review]
LGTM

**Pull request merged**